src/scripts/generate-heading-ids.ts

```diff
@@ -13,16 +13,27 @@ const GitHubSlugger = require("github-slugger")
 
 // TODO we should auto-run this script when new markdown files are added to the project
 
-const toc = {}
+interface IGitHubSlugger {
+  slug: (headingText: string) => string
```

Comment:
refactor: I should install `@types/github-slugger`

Reply:
@pettinarip 

Are Definitely Typed types detected with a CommonJS require syntax?
I can use `typeof gitHubSlugger` to specify its type, but, can't use `BananaSlug` which is the [name of its type class definition from the @types/github-slugger package](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/github-slugger/index.d.ts#L10)

Reply:
Or an alternate solution: declare the script as ES6 Module and use the `import` syntax.

Reply:
Yes, lets try that. Lets use it as an es6 module where you can do these imports using `import`.

Reading a bit, I think you will need to override the tsconfig for `ts-node` to use commonjs module, like this:
```
// in tsconfig.json add
"ts-node": {
  "compilerOptions": {
    "module": "commonjs"
  }
}
```

Reply:
@pettinarip 

For now, I am using the `.mts` extension but, if you want then, I think [we can use `moduleTypes`](https://typestrong.org/ts-node/docs/module-type-overrides) to define the module type for this script explicitly.

Reply:
Also, since module type is inferred from the nearest package.json, we can add a `package.json` in the scripts folder and [specify the type as `module` for all the scripts](https://nodejs.org/docs/latest-v13.x/api/esm.html#esm_enabling).

But, this option would require converting the other scripts to use the ESM import and will go out of the scope of this PR.

Reply:
Yes, I guess I like to have everything in `.ts` instead of mixing them with different extensions to make it simpler to other developers. But I don't have a strong opinion about that, I think we are fine with using it as `.mts`.